test/ERC6538Registry.t.sol

```diff
@@ -47,24 +67,296 @@ contract RegisterKeys is ERC6538RegistryTest {
   /// forge-config: default.fuzz.runs = 1
   /// forge-config: ci.fuzz.runs = 1
   /// forge-config: lite.fuzz.runs = 1
-  function testFuzz_NeverReverts(address caller, uint256 schemeId, bytes memory stealthMetaAddress)
-    external
-  {
+  function testFuzz_AlwaysSucceeds(
+    address caller,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress
+  ) external {
     vm.prank(caller);
     registry.registerKeys(schemeId, stealthMetaAddress);
   }
 }
 
-contract RegisterKeysOnBehalf_Address is ERC6538RegistryTest {
-  function test_NotImplemented() external {
-    vm.expectRevert("not implemented");
+contract RegisterKeysOnBehalf is ERC6538RegistryTest {
+  function testFuzz_Erc712SignatureIsValid(
+    string memory name,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress
+  ) external {
+    (address alice, uint256 alicePk) = makeAddrAndKey(name);
+    SigUtils.RegistrantInfo memory registrantInfo =
+      SigUtils.RegistrantInfo(schemeId, stealthMetaAddress, 0 /* nonce */ );
+    bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+    bytes memory signature = abi.encodePacked(r, s, v);
+
+    vm.expectEmit(true, true, true, true);
+    emit StealthMetaAddressSet(alice, schemeId, stealthMetaAddress);
+    registry.registerKeysOnBehalf(alice, schemeId, signature, stealthMetaAddress);
+  }
+
+  function testFuzz_Erc1271SignatureIsValid(
+    string memory name,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress
+  ) external {
+    (address alice, uint256 alicePk) = makeAddrAndKey(name);
+
+    vm.prank(alice);
+    ERC1271MockContract erc1271MockContract = new ERC1271MockContract();
+    address registrant = address(erc1271MockContract);
+
+    SigUtils.RegistrantInfo memory registrantInfo =
+      SigUtils.RegistrantInfo(schemeId, stealthMetaAddress, 0 /* nonce */ );
+    bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+    bytes memory signature = abi.encodePacked(r, s, v);
+
+    vm.expectEmit(true, true, true, true);
+    emit StealthMetaAddressSet(registrant, schemeId, stealthMetaAddress);
+    registry.registerKeysOnBehalf(registrant, schemeId, signature, stealthMetaAddress);
+  }
+
+  function testFuzz_UpdateStealthMetaAddress(
```

Comment:
Similar changes are needed here to produce useful spec outputs and to ensure the contract state has actually been updated.

Reply:
Updated in [3a5aa7c](https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/3a5aa7ce082f3580802dcad5e30884c9a736e04a)